hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/ozoneimpl/ContainerScanHelper.java

```diff
@@ -107,6 +109,18 @@ public void handleUnhealthyScanResult(long containerID, ScanResult result) throw
     }
   }
 
+  public void triggerVolumeScan(ContainerData containerData) {
+    HddsVolume volume = containerData.getVolume();
+    if (volume != null && !volume.isFailed()) {
+      log.info("Triggering a volume scan for volume [{}] as unhealthy container [{}] was on it.",
+          volume.getStorageDir().getPath(), containerData.getContainerID());
+      StorageVolumeUtil.onFailure(volume);
+    } else {
+      log.warn("Cannot trigger volume scan for container {} since its volume is null or has failed.",
```

Comment:
Can we add different logs for each case? Null case sounds like its alerting to a possible bug, but failed case is not.